dist/crypto.js

```diff
@@ -0,0 +1,291 @@
+  * @returns {arrayBuffer|Object} - The raw key or the key as a jwk format
+  */
+var exportKey = function exportKey(key) {
+  var type = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'raw';
+
+  return window.crypto.subtle.exportKey(type, key).then(function (key) {
+    if (type === 'raw') return new Uint8Array(key);
+    return key;
+  });
+};
+
+/**
+  * Import a raw|jwk as a CryptoKey
+  *
+  * @param {arrayBuffer|Object} key - The key
+  * @param {string} [type] - The type of the key to import ('raw', 'jwk')
+  * @param {string} [mode] - The mode of the key to import (default 'AES-GCM')
+  * @returns {arrayBuffer|Object} - The cryptoKey
+  */
+var importKey = function importKey(key) {
+  var type = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'raw';
+  var mode = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 'AES-GCM';
+
+  return window.crypto.subtle.importKey(type, key, { name: mode }, true, ['encrypt', 'decrypt']);
+};
+
+/**
+ * Decrypt buffer
+ *
+ * @param {ArrayBuffer} data - Data to decrypt
+ * @param {ArrayBuffer} key - The AES key as raw data. 128 or 256 bits
+ * @param {Object} cipherContext - The AES cipher parameters
+ * @param {ArrayBuffer} cipherContext.iv - The IV
+ * @param {string} cipherContext.name - The encryption mode
+ * @param {ArrayBuffer} [cipherContext.additionalData] - The non-secret authenticated data (only AES-GCM)
+ * @param {ArrayBuffer} [cipherContext.counter] - The counter used for aes-ctr mode
+ * @returns {ArrayBuffer} - The decrypted buffer
+ */
+var decryptBuffer = function decryptBuffer(data, key, cipherContext) {
+  // TODO: test input params
+  return window.crypto.subtle.decrypt(cipherContext, key, data).then(function (result) {
+    return new Uint8Array(result);
+  });
+};
+
+/**
+ * Encrypt buffer
+ *
+ * @param {ArrayBuffer} data - Data to encrypt
+ * @param {ArrayBuffer} key - The AES CryptoKey
+ * @param {Object} cipherContext - The AES cipher parameters
+ * @param {ArrayBuffer} cipherContext.iv - The IV
+ * @param {string} cipherContext.name - The encryption mode
+ * @param {ArrayBuffer} [cipherContext.additionalData] - The non-secret authenticated data (only AES-GCM)
+ * @param {ArrayBuffer} [cipherContext.counter] - The counter used for aes-ctr mode
+ * @returns {ArrayBuffer} - The encrypted buffer
+ */
+var encryptBuffer = function encryptBuffer(data, key, cipherContext) {
+  return window.crypto.subtle.encrypt(cipherContext, key, data).then(function (result) {
+    return new Uint8Array(result);
+  });
+};
+
+/**
+ * Encrypt data
+ *
+ * @param {CryptoKey} key - The AES CryptoKey
+ * @param {string | Object} - The data to encrypt
+ * @param {string} [format] - The ciphertext and iv encoding format
+ * @returns {Object} - The stringified ciphertext object (ciphertext and iv)
+ */
+var encrypt = function encrypt(key, data) {
+  var format = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 'hex';
```

Comment:
Why not add format as a parameter of the encrypt funciton ?

Reply:
Beware, its a dist file ahah

Reply:
base64 or hex encoding of iv and ciphertext.

Reply:
:man_facepalming: